apps/help-center/help-center-gutenberg.js

```diff
@@ -6,61 +6,74 @@ import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
 import { Button, Fill } from '@wordpress/components';
 import { useMediaQuery } from '@wordpress/compose';
 import { useDispatch, useSelect } from '@wordpress/data';
-import { useCallback, useEffect, useState } from '@wordpress/element';
+import { useCallback, useEffect, useState, useReducer } from '@wordpress/element';
 import { registerPlugin } from '@wordpress/plugins';
+import ReactDOM from 'react-dom';
+import { useCanvasMode } from './hooks';
 import './help-center.scss';
 
 const queryClient = new QueryClient();
 
 function HelpCenterContent() {
-	const [ helpIconRef, setHelpIconRef ] = useState();
+	const [ , forceUpdate ] = useReducer( ( x ) => x + 1, 0 );
 	const isDesktop = useMediaQuery( '(min-width: 480px)' );
 	const [ showHelpIcon, setShowHelpIcon ] = useState( false );
 	const { setShowHelpCenter } = useDispatch( 'automattic/help-center' );
 
 	const show = useSelect( ( select ) => select( 'automattic/help-center' ).isHelpCenterShown() );
 
+	const canvasMode = useCanvasMode();
+
 	const handleToggleHelpCenter = useCallback( () => {
 		recordTracksEvent( `calypso_inlinehelp_${ show ? 'close' : 'show' }`, {
 			force_site_id: true,
 			location: 'help-center',
 			section: 'gutenberg-editor',
+			canvas_mode: canvasMode,
 		} );
 
 		setShowHelpCenter( ! show );
-	}, [ setShowHelpCenter, show ] );
+	}, [ setShowHelpCenter, show, canvasMode ] );
 
 	useEffect( () => {
 		const timeout = setTimeout( () => setShowHelpIcon( true ), 0 );
 		return () => clearTimeout( timeout );
 	}, [] );
 
+	useEffect( () => {
+		// Close the Help Center when the canvas mode changes.
+		setShowHelpCenter( false );
+
+		// Force to re-render to ensure the sidebar is available.
+		if ( canvasMode === 'view' ) {
+			forceUpdate();
+		}
```

Comment:
Isn't `setShowHelpCenter( false );` sufficient to re-render here?

Reply:
It doesn't work if it's closed

Reply:
You're right, I misunderstood this effect.